taipy/core/config/checkers/_data_node_config_checker.py

```diff
@@ -196,28 +196,43 @@ def _check_generic_read_write_fct_and_args(self, data_node_config_id: str, data_
                     f"DataNodeConfig `{data_node_config_id}` must be populated with a Callable function.",
                 )
 
-    def _check_callable(self, data_node_config_id: str, data_node_config: DataNodeConfig):
-        properties_to_check = {
+    @staticmethod
+    def _get_class_type_and_properties() -> Dict[str, List[Tuple[Any, List[str]]]]:
```

Comment:
this can be turned into a class attribute instead of a function, as we were using function, I just went with it so can quickly implement it

Reply:
I agree. We can implement it as a dictionary class attribute of the `DataNoceConfig` class. Just like we have `_REQUIRED_PROPERTIES: Dict[str, List]` or `_OPTIONAL_PROPERTIES: Dict[str, Dict[str, Any]]`, we could have something like :
```python
_PROPERTIES_TYPES: Dict[str, Dict[str, type]] = {
    DataNodeConfig._STORAGE_TYPE_VALUE_GENERIC: {
        DataNodeConfig._OPTIONAL_READ_FUNCTION_GENERIC_PROPERTY: Callable,
        DataNodeConfig._OPTIONAL_WRITE_FUNCTION_GENERIC_PROPERTY: Callable},
    DataNodeConfig._STORAGE_TYPE_VALUE_SQL: {
        DataNodeConfig._REQUIRED_WRITE_QUERY_BUILDER_SQL_PROPERTY: Callable,
        DataNodeConfig._OPTIONAL_APPEND_QUERY_BUILDER_SQL_PROPERTY: Callable}
}
```
By the way, we could use it to check the types of all properties (both required and optional), not only for callables.
Then, this dictionary could be extended to enterprise.

Does it make sense?

Reply:
Yep, that's my initial intention with this refactoring. Trying to reuse the code for different types of instances. Though I'd say we should find a different ticket for that 😅